components/global/mixer-adapter/cmd/mixer-adapter/main.go

```diff
@@ -58,46 +100,123 @@ func main() {
-	   This is optional. */
-	adapterCertificate := os.Getenv(grpcAdapterCertificatePath) // adapter.crt //change the name
-	adapterprivateKey := os.Getenv(grpcAdapterPrivateKeyPath)   // adapter.key
-	caCertificate := os.Getenv(caCertificatePath)               // ca.pem
-	spServerUrl := os.Getenv(spServerUrlPath)
-
-	logger.Infof("Sp server url : %s", spServerUrl)
+	// Mutual TLS feature to secure connection between workloads. This is optional.
+	adapterCertificate := config.Mixer.TLS.Certificate // adapter.crt
+	adapterPrivateKey := config.Mixer.TLS.PrivateKey   // adapter.key
+	caCertificate := config.Mixer.TLS.CaCertificate    // ca.pem
+	// Initialize the variables from the config map
+	spServerUrl := config.SpEndpoint.URL
+	filePath := config.Store.FileStorage.Path
+	bufferTimeoutSeconds := config.Advanced.BufferTimeoutInSeconds
+	minBufferSize := config.Advanced.BufferSize
+	tickerSec := config.SpEndpoint.SendIntervalSeconds
 
 	client := &http.Client{}
-	publisher := adapter.SPMetricsPublisher{}
-
 	var serverOption grpc.ServerOption = nil
-
 	if adapterCertificate != "" {
-		serverOption, err = getServerTLSOption(adapterCertificate, adapterprivateKey, caCertificate)
+		serverOption, err = getServerTLSOption(adapterCertificate, adapterPrivateKey, caCertificate)
 		if err != nil {
 			logger.Warn("Server option could not be fetched, Connection will not be encrypted")
 		}
 	}
-
-	spAdapter, err := adapter.New(port, logger, client, publisher, serverOption, spServerUrl)
+	buffer := make(chan string, minBufferSize*config.Advanced.BufferSizeFactor)
+	errCh := make(chan error, 1)
+	spAdapter, err := adapter.New(port, logger, client, serverOption, spServerUrl, buffer)
 	if err != nil {
-		logger.Fatal("unable to start server: ", err.Error())
+		logger.Fatalf("unable to start the server: ", err.Error())
+	}
+	if spAdapter == nil {
+		logger.Fatal("adapter.New returned a null struct")
+		return
+	}
+	go spAdapter.Run(errCh)
+	var ps store.Persister
+	wrt := &writer.Writer{
+		WaitingTimeSec: bufferTimeoutSeconds,
+		WaitingSize:    minBufferSize,
+		Logger:         logger,
+		Buffer:         buffer,
+		StartTime:      time.Now(),
+	}
+	ticker := time.NewTicker(time.Duration(tickerSec) * time.Second)
+	pub := &publisher.Publisher{
+		Ticker:      ticker,
+		Logger:      logger,
+		SpServerUrl: spServerUrl,
+		HttpClient:  &http.Client{},
+	}
+
+	var metricsStore = config.Store
+	// Check the config map to initialize the correct persistence
+	if metricsStore.FileStorage.Path != "" {
+		// File storage will be used for persistence. Priority will be given to the file system
+		logger.Info("Enabling file persistence")
+		ps = &file.Persister{
+			Logger:    logger,
+			Directory: filePath,
+		}
+	} else if (metricsStore.Database.Host != "") && (metricsStore.Database.Username != "") {
+		// Db will be used for persistence
+		logger.Info("Enabling database persistence")
+		dsn := (&mysql.Config{
```

Comment:
AFAIR we discussed to change the variable name to something more intuitive in the last review.